data/src/server.rs

```diff
@@ -9,30 +9,73 @@ use serde::{Deserialize, Serialize};
 use tokio::fs;
 use tokio::process::Command;
 
-use crate::config;
+use crate::bouncer::BouncerNetwork;
 use crate::config::Error;
 use crate::config::server::Sasl;
+use crate::config::sidebar::OrderBy;
+use crate::config::{self, sidebar};
 
 pub type Handle = Sender<proto::Message>;
 
+pub type ServerName = Arc<str>;
+
 #[derive(Debug, Clone, Hash, PartialEq, Eq, Serialize, Deserialize)]
-pub struct Server(Arc<str>);
+pub struct Server {
+    pub name: ServerName,
+    pub network: Option<Arc<BouncerNetwork>>,
+}
+impl Server {
+    pub fn is_bouncer_network(&self) -> bool {
+        self.network.is_some()
+    }
+
+    pub fn bouncer_netid(&self) -> Option<&str> {
+        self.network.as_ref().map(|network| &*network.id)
+    }
+
+    pub fn parent(&self) -> Option<Server> {
+        if self.network.is_some() {
+            Some(Self {
+                network: None,
+                ..self.clone()
+            })
+        } else {
+            None
+        }
+    }
+}
 
-impl From<&str> for Server {
-    fn from(value: &str) -> Self {
-        Server(Arc::from(value))
+impl From<ServerName> for Server {
+    fn from(name: ServerName) -> Self {
+        Self {
+            name,
+            network: None,
+        }
     }
 }
 
 impl fmt::Display for Server {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.0.fmt(f)
+        if let Some(network) = &self.network {
+            network.name.fmt(f)
+        } else {
+            self.name.fmt(f)
+        }
```

Comment:
`Display` drives some hashing for file access & hashmap key lookups, we probably don't want an accidental collision between a similarly named config network & bouncer network causing collisions. We should concat bouncer name w/ network name to ensure its unique, similar to how you do it for the binary format

Reply:
I would rather sniff these out and fix them, I tried to do that in this PR (and I've been daily driving it for a few weeks so i think I've taken care of most of them)